core/util_test.go

```diff
@@ -433,13 +433,16 @@ func (d *DrandTestScenario) SetMockClock(t *testing.T, targetUnixTime int64) {
 
 // AdvanceMockClock advances the clock of all drand by the given duration
 func (d *DrandTestScenario) AdvanceMockClock(t *testing.T, p time.Duration) {
+	t.Log("Advancing time by", p, "from", d.clock.Now().Unix())
 	for _, node := range d.nodes {
 		node.clock.Advance(p)
 	}
 	for _, node := range d.newNodes {
 		node.clock.Advance(p)
 	}
 	d.clock.Advance(p)
+	// we sleep to make sure everyone has the time to get the new time before continuing
+	time.Sleep(10 * time.Millisecond)
```

Comment:
should we advance the the clock as well?

Reply:
No, it's a sleep meant to allow the other thread to "catch up" on the fake time clock advancement.